apps/aehttp/src/aehttp_dispatch_int.erl

```diff
@@ -210,15 +210,21 @@ handle_request('PostNameClaimTx', #{'NameClaimTx' := NameClaimTxObj}, _Context)
             #{<<"name">>      := Name,
               <<"name_salt">> := NameSalt,
               <<"fee">>       := Fee} = NameClaimTxObj,
-            {ok, ClaimTx} =
-                aens_claim_tx:new(
-                  #{account   => PubKey,
-                    nonce     => Nonce,
-                    name      => Name,
-                    name_salt => NameSalt,
-                    fee       => Fee}),
-            sign_and_push_to_mempool(ClaimTx),
-            {200, [], #{name_hash => aec_base58c:encode(name, aens_hash:name_hash(Name))}};
+            case aens:get_name_hash(Name) of
+                {ok, NameHash} ->
+                    {ok, ClaimTx} =
+                        aens_claim_tx:new(
+                          #{account   => PubKey,
+                            nonce     => Nonce,
+                            name      => Name,
+                            name_salt => NameSalt,
+                            fee       => Fee}),
+                    sign_and_push_to_mempool(ClaimTx),
+                    {200, [], #{name_hash => aec_base58c:encode(name, NameHash)}};
+                {error, Reason} ->
+                    ReasonBin = atom_to_binary(Reason, utf8),
```

Comment:
Same here. Not critical as it is error path anyway though user would see 500 rather than 400. And 400 without reason message is better than 500.

Reply:
Spec changed to `term()`